awsx/ec2/vpc.ts

```diff
@@ -157,6 +105,29 @@ export class Vpc extends schema.Vpc<VpcData> {
     );
     const vpcId = vpc.id;
 
+    // If the user did not specify cidrBlock, use the computed value allocated by the VPC.
+    const actualCidrBlock: pulumi.Input<string> = cidrBlock ?? vpc.cidrBlock;
+
+    const { subnetSpecs: decidedSpecs, subnetLayout } = this.decideSubnetSpecs(
+      name,
+      actualCidrBlock,
+      subnetStrategy,
+      availabilityZones,
+      args,
+    );
+
+    const subnetSpecs = validatePartialSubnetSpecs(decidedSpecs, (subnetSpecs) => {
+      validateSubnets(subnetSpecs, getOverlappingSubnets);
```

Comment:
Hmm, self-reviewing this change, there's a problem here that validations now happen after the new aws.ec2.Vpc call, and in the case when they used to terminate the flow promptly, that's a visible change, as now `new aws.ec2.Vpc` attempts to provision before we reject the validations. It's a bit of a chicken and egg problem.

Reply:
I think it's the best that we can do. From the user's perspective I would have to have created the VPC resource in order to have the cidr allocated.

How does pulumi run the validation in this case? Are the creation of the other resources dependent on the validation completing or would they be run in parallel?

Reply:
It runs after the internal vpc resource registers but before subnets do. If there's error reported by the internal vpc resource they now will be reported in addition to validation failures reported by awsx.Vpc. At a cost of a little bit of extra code complexity we can get the previous behavior back while using the new validation ordering behavior for the IPAM case.

Reply:
Yeah pulumi runs resource allocations in parallel, so when we say new Vpc() we fork off a thread creating that VPC. Validations can run in parallel or they can block resource creation.